code/frontend/src/components/UserProfile.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+function UserProfile() {
+  return (
+    <div>
+      <head>
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>User Profile</title>
+        <link rel="stylesheet" href="../styles/UserProfile.css" type="text/css" media="all" />
+      </head>
+      <body>
+        <header>
+          {/* Home Button */}
+          <div className="home-container">
+            <button id="home">🔄 Home</button>
+          </div>
+
+          {/* User Profile Container */}
+          <div className="profile-container">
+            <div className="profile-pic">
+              <div className="profile-pic">
+                <img
+                  src="https://www.shareicon.net/data/256x256/2016/09/15/829472_man_512x512.png"
+                  alt="Profile Image"
+                />{' '}
+                {/* Replace with your image */}
+              </div>
+            </div>
+            <span>Supriya Uppala</span>
+          </div>
+          <span></span> {/* Placeholder element */}
+        </header>
+
+        <section className="hobbies">
+          <h2>Hobbies:</h2>
+          <button>Singing</button>
```

Comment:
This is hardcoded. the problem is, if I were to add another hobby to it then it won't show on this page because it's hardcoded to only have 3 hobbies. 
Can you update this to loop through a list of hobbies in an array? 
For example if I have an array like this ["Singing", "Dancing", "Hiking"]
We should have 3 buttons for each element in the loop.